ceno_rt/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+#![deny(clippy::cargo)]
+#![feature(strict_overflow_ops)]
+#![feature(linkage)]
+
+#[cfg(target_arch = "riscv32")]
+use core::arch::{asm, global_asm};
+
+mod allocator;
+
+mod mmio;
+pub use mmio::{read, read_slice};
+
+mod io;
+pub use io::info_out;
+
+mod params;
+pub use params::*;
+
+#[no_mangle]
+#[linkage = "weak"]
+pub extern "C" fn sys_write(_fd: i32, _buf: *const u8, _count: usize) -> isize {
+    unimplemented!();
+}
+
+pub fn halt(exit_code: u32) -> ! {
+    #[cfg(target_arch = "riscv32")]
+    unsafe {
+        asm!(
+            "ecall",
+            in ("a0") exit_code,
+            in ("t0") 0,
+        );
+        unreachable!();
+    }
+    #[cfg(not(target_arch = "riscv32"))]
+    unimplemented!("Halt is not implemented for this target: {}", exit_code);
```

Comment:
`unimplemented!("Halt is not implemented for this target, exit_code: {}", exit_code);` would be easier to understand

Reply:
Good idea!